web/store/inbox/inbox-issue.store.ts

```diff
@@ -151,4 +158,23 @@ export class InboxIssueStore implements IInboxIssueStore {
       });
     }
   };
+
+  updateProjectIssue = async (issue: Partial<TIssue>) => {
+    const inboxIssue = this.issue;
+    try {
+      if (!this.issue.id) return;
+      Object.keys(issue).forEach((key) => {
+        const issueKey = key as keyof TIssue;
+        set(inboxIssue, issueKey, issue[issueKey]);
+      });
+      await this.issueService.patchIssue(this.workspaceSlug, this.projectId, this.issue.id, issue);
+      // fetching activity
+      await this.store.issue.issueDetail.fetchActivities(this.workspaceSlug, this.projectId, this.issue.id);
+    } catch {
+      Object.keys(issue).forEach((key) => {
```

Comment:
If you are trying to revert back the state if the API call fails, this wont work. You'll need to clone the issue so that it changes reference and store it as initial data for this to work. And use that initial data to revert back the change. Please check